helm-chart/templates/01-service-account.yaml

```diff
@@ -2,14 +2,11 @@
 apiVersion: v1
 kind: ServiceAccount
 metadata:
-  creationTimestamp: null
   labels:
-  {{- if .Values.tap.labels }}
-    {{- toYaml .Values.tap.labels | nindent 4 }}
-  {{- end }}
+    {{- include "kubeshark.labels" . | nindent 4 }}
   annotations:
   {{- if .Values.tap.annotations }}
     {{- toYaml .Values.tap.annotations | nindent 4 }}
   {{- end }}
-  name: kubeshark-service-account
+  name: {{ include "kubeshark.serviceAccountName" . }}
```

Comment:
What is the motivation to template the service account name?

Reply:
Templating the Service Account name in a Helm chart brings consistency, automation, scalability, and adaptability to the deployment process. It aligns with best practices for resource management, enhances security through naming conventions, and supports the efficient management of resources across diverse environments.  it's important to carefully consider the potential drawbacks in terms of complexity, standardization, compatibility, debugging, flexibility, maintenance, and the learning curve.

Reply:
- A company that requires that every service account have the name of the team on it
- A policy by a third-party team that restricts access based on a pattern in the names

I saw those two things happen in different companies. In both cases, the team requesting a tool to be installed wasn't able to change/suggest internal policies. The question is why the user couldn't change the default name.

Reply:
I understood, thanks!